repeatLandsatWidths.js

```diff
@@ -0,0 +1,250 @@
+  
+  var wm = ee.Image(0); // water mask
+  wm = wm.where(cm.eq(1), 1);
+  cm = cm.where(cm.eq(1), 0);
+
+  //wm = wm.clip(image.geometry()); // fix me: clip buffered, not the mask.
+  //wm = wm.updateMask(wm);
+  //cm = cm.updateMask(cm);
+  //var intersection = eeBuffered.intersection(ee.Feature(image.geometry()));
+  //var clpd = mask.clip(buffered);  // clip mask to buffered features for display
+  
+  // http://stackoverflow.com/questions/2666112/convert-miliseconds-to-date-in-excel
+  var date = ee.Number(image.get('system:time_start'));
+  
+  // Add bands to the wm
+  wm = wm.addBands([date, cm])
+            .rename(["water", "date", "cloud"]);
+  
+  return(wm);
+  
+});
+
+
+
+
+
+
+////////////////////////////////////////////////////////////////////////
+// take average of masks in buffered cross sections:
+var maskTab = maskCollection.map(function(image){
+  var reduced = image.reduceRegions({
+                                      collection: buffered,
+                                      reducer: ee.Reducer.mean(),
+                                      scale: 30
+                                    });
+  return reduced;
+});
+
+maskTab = maskTab.flatten();
+
+
+
+
+
+
+
+////////////////////////////////////////////////////////////////////////
+// export table to drive:
+
+// keep only essential properties in table: 
+var maskTab = maskTab.map(function(f) {
+                            return ee.Feature(f.select(
+                              ["segmentID", "segmentInd", "date", 
+                              'water', "cloud", "xSecLength"]
+                              , null, false));
+                          });
+
+
+//console.log(maskTab);
+
+// fixme: multiply properties to get width property:
+// var maskTab = maskTab.map(function(f) {
+//                             return ee.Feature(f.select(['water']);
+//                           });
+
+
+// var water = ee.List(maskTab.aggregate_array('water'));
+// var xSecLength = ee.List(maskTab.aggregate_array('xSecLength'));
+// var width = water.toArray().divide(xSecLength.toArray());
+// var width = water.zip(xSecLength).map(function(f) { return ee.Number(f.get(0)).divide(f.get(1))});
+
+
+Export.table(maskTab, 'AMHG_LT5_1', 
+            {fileFormat: 'CSV',
+              driveFolder: 'GEE',
+              driveFileNamePrefix: 'AMHG_LT5_1'
+            });
+
+
+////////////////////////////////////////////////////////////////////////
+//Export mask video to drive:
+// var vid = wmMn
+//     .select(['clpd', 'nir', 'green'])
+//     .map(function(image) {
+//       return  image.multiply(512).uint8();
+//     });
+
+
+
+// Export.video(vid, 'sinTile_masks_vid01', 
+//             {dimensions: '720',
+//             framesPerSecond: 4,
+//             region: ee.Geometry.Rectangle([-104.27, 48.07, -104.08, 47.98]).toGeoJSON(),
+//             driveFolder: 'GEE',
+//             driveFileNamePrefix: 'waterVideo_v01'
+//             });
+
```

Comment:
Sometimes rather than comment out code you are not using, you can use something like

     var exportVideo = false
     if (exportVideo) {
       ...
     }

then, obviously, to enable it you just change false to true. Or you can easily change it to rely on an input argument to set the exportVideo flag.